### PR TITLE
feat(auth): support enterprise Teams endpoints

### PR DESCRIPTION
**Pull request by gekiclaws**
## Problem

The bridge assumes consumer Teams authentication formats and consumer chat-service endpoints. Organizational accounts return a different Skype-token envelope and regional service URLs, so their tokens cannot be parsed and routed reliably.

Impact is not yet quantified. This is foundational support for organizational authentication; tenant/account selection remains separate work.

## Solution

Add enterprise Skype-token parsing, including JWT-based Skype ID recovery, business-token fields, and regional service discovery. Extract Skype API-scoped access tokens from MSAL caches and route chat operations through the persisted regional endpoint.

Keep durable personal device-code logins pinned to their issuing client and consumer endpoints. Advanced endpoint and scope overrides apply only to legacy or externally provisioned organizational logins, and regional chat URLs must be absolute HTTPS URLs.

The fork's broad connector refactor and manual localStorage flow were intentionally not adopted; this PR ports only the authentication and routing foundation.

## Testing

- [ ] Manual testing steps performed
- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Tested edge cases: consumer and enterprise token envelopes, missing Skype IDs recovered from JWT claims, newest enterprise access-token selection, unsafe regional URLs, persisted issuing-client precedence, and personal-login isolation from enterprise overrides

## Risks

Teams' enterprise chat endpoints are not a stable public API, so response shapes or regional routing conventions may change. Configuration is bridge-wide for now; multi-tenant account selection and per-login tenant configuration remain follow-up work.

## Screenshots/Videos

No UI changes

## Related

Fixes #[TODO: add issue number]
Depends on #[TODO: add prerequisite PR number]

---

## Reviewer Notes

Please focus on the dual consumer/enterprise Skype-token parsing, the HTTPS regional endpoint normalization, and whether the global organizational refresh overrides are sufficiently isolated from personal device-code logins.

### CODE REVIEW

File: internal/teams/client/endpoints.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		return errors.New("teams client is nil")
+	}
+	baseURL, err := normalizeChatServiceURL(rawURL)
+	if err != nil {
+		return err
+	}
+	c.ConversationsURL = baseURL + "/v1/users/ME/conversations"
+	c.MessagesURL = baseURL + "/v1/users/ME/conversations"
+	c.SendMessagesURL = baseURL + "/v1/users/ME/conversations"
+	c.ConsumptionHorizonsURL = baseURL + "/v1/users/ME/threads"
```
Comment (ransomts) [inline internal/teams/client/endpoints.go]: The regional horizons path has no `users/ME` segment; this form returned 404 on my tenant and `/v1/threads/...` returned 200 (details in the review comment).

```suggestion
	c.ConsumptionHorizonsURL = baseURL + "/v1/threads"
```

**Review by ransomts — COMMENTED** [review]
I maintain an enterprise fork (ransomts/mautrix-teams) and use it every day against a university tenant with Conditional Access, so I can check this against a real organizational account. Notes from reading the diff against what works for me:

**1. Consumption horizons URL returns 404** (suggestions inline). `ConfigureChatServiceURL` builds `{chatService}/v1/users/ME/threads/{id}/consumptionhorizons`. I tried both forms against my tenant (`amer.ng.msg.teams.microsoft.com`, two chats, same skypetoken):

| Path | Result |
|---|---|
| `/v1/users/ME/threads/{id}/consumptionhorizons` | `404 {"errorCode":404,…"errorSubCode":1}` |
| `/v1/threads/{id}/consumptionhorizons` | `200`, `consumptionhorizons` for both members |

So inbound read receipts would fail on enterprise. Outbound ones are fine, since they go through `SendMessagesURL`. The test asserts the `users/ME` form, which is why it passes.

**2. No way to create an enterprise login.** The only login flow is personal device code, and `StartHelperListener` / `ExtractTokensFromMSALLocalStorage` have no caller in `pkg/`. So the new `selectSkypeAPIAccessToken` path is unreachable, and "externally provisioned" logins have no entry point. Also, device code isn't an option on many enterprise tenants: Conditional Access blocks non-browser sign-in (AADSTS 53003) on mine. What works for me is the Teams web app's own auth-code + PKCE flow in a real browser, with the tokens submitted through the provisioning API. Is there a login shape you'd accept for that, e.g. a "tokens" flow that takes a refresh token plus its client ID and scope? I'd be glad to write it to fit how you want it.

**3. The regional AMS URL is parsed but not persisted.** `SkypeTokenResult.AMSURL` is populated and then dropped. On my tenant, inline images needed both the regional AMS host and the header `Authorization: skype_token <token>` (not the consumer `authentication: skypetoken=…`); without them, image downloads failed.

**4. Refresh scope per login rather than bridge-wide.** `skype_refresh_scopes` is global. I store the scope with the login, the same way you already store `OAuthClientID`, so a refresh always matches what the login was issued with. It also keeps personal and enterprise logins on the same bridge from interfering with each other.

**5. Long-poll (not part of this diff).** In case it helps with further enterprise work: the enterprise chat service returns 410 Gone when registering the HTTP long-poll endpoint (both v1 POST and v2 PUT). I replaced it with a 3-second poll of the 30 most recent conversations, which wakes the threads whose `lastMessage.id` changed. I can send that as a separate PR once enterprise auth has landed.

The metadata key `region_chat_service_url` matches mine, so existing logins would carry over. Once there's a login path for organizational accounts, I'm happy to run this branch on my tenant and report back.

Thank you for building and maintaining this bridge. Attachments, reactions, receipts, and the steady fixes since made it possible for me to use Teams from my own Matrix client at all, and it's good to see enterprise support on the way.

File: internal/teams/client/endpoints_test.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	if err != nil {
+		t.Fatalf("configure regional endpoint: %v", err)
+	}
+	if client.ConversationsURL != "https://amer.ng.msg.teams.microsoft.com/v1/users/ME/conversations" {
+		t.Fatalf("unexpected conversations URL: %q", client.ConversationsURL)
+	}
+	if client.MessagesURL != client.ConversationsURL || client.SendMessagesURL != client.ConversationsURL {
+		t.Fatalf("message endpoints were not routed together: %+v", client)
+	}
+	if client.ConsumptionHorizonsURL != "https://amer.ng.msg.teams.microsoft.com/v1/users/ME/threads" {
```
Comment (ransomts) [inline internal/teams/client/endpoints_test.go]:
```suggestion
	if client.ConsumptionHorizonsURL != "https://amer.ng.msg.teams.microsoft.com/v1/threads" {
```
